graphql/project_settings_resolver.go

```diff
@@ -16,6 +18,22 @@ func (r *projectSettingsResolver) Aliases(ctx context.Context, obj *restModel.AP
 	return getAPIAliasesForProject(ctx, utility.FromStringPtr(obj.ProjectRef.Id))
 }
 
+// GithubAppAuth is the resolver for the githubAppAuth field.
+func (r *projectSettingsResolver) GithubAppAuth(ctx context.Context, obj *restModel.APIProjectSettings) (*restModel.APIGithubAppAuth, error) {
+	app, err := model.FindOneGithubAppAuth(utility.FromStringPtr(obj.ProjectRef.Id))
+	if err != nil {
+		return nil, InternalServerError.Send(ctx, fmt.Sprintf("finding GitHub app for project '%s': %s", utility.FromStringPtr(obj.ProjectRef.Id), err.Error()))
+	}
+	// It's valid for a project to not have a Github app defined.
+	if app == nil {
+		return nil, nil
+	}
+	res := &restModel.APIGithubAppAuth{}
+	app = app.RedactPrivateKey()
```

Comment:
Re my comment about redacting from the API, if we're entirely redacting it from the REST API and GraphQL, rather than redacting manually, could we just have the private key be sent as an empty/nonexistent string, and the UI just displays that as redacted? Would avoid having to remember to redact the key by making the key write-only (and not readable).